### PR TITLE
add night.design

### PR DESCRIPTION
**Pull request by Nightcorelena**

### CODE REVIEW

File: websites/N/Night.design/metadata.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	"author": {
+		"name": "BassKatze",
+		"id": "766213190332710913"
+	},
+	"contributors": [
+		{
+			"name": "Lena the fauli",
+			"id": "241278257335500811"
+		}
+	],
```
Comment (EncryptedDev) [inline websites/N/Night.design/metadata.json:12]: A contributor must have contributed to at least 20% of the codebase, as this is a new presence there cannot be a contributor so far.

File: websites/N/Night.design/metadata.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	"service": "night.design",
+	"description": {
+        "en": "Night.design is a collection of textures like capes, bandanas and shields to use in Minecraft."
+	},
```
Comment (EncryptedDev) [inline websites/N/Night.design/metadata.json:16]: Poor formatting, use the script to format.

File: websites/N/Night.design/presence.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    presenceData.buttons = [
+        {
+            label: "View Website",
+            url: "https://night.design/"
+        },
+        {
+            label: "Actually Page",
+            url: document.location.href
+        }
+    ]
```
Comment (EncryptedDev) [inline websites/N/Night.design/presence.ts]: Static pages are not allowed

File: websites/N/Night.design/presence.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+    let domain = document.location.pathname.slice(3);
```
Comment (EncryptedDev) [inline websites/N/Night.design/presence.ts]: `domain` is a static variable so should use `const` instead

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
`if` and `else-if`s should be put in a `Record` if they're all static

File: websites/N/Night.design/metadata.json (hunk 1 above)
Comment (Nightcorelena) [reply websites/N/Night.design/metadata.json:12]: But i have working on it as a dev but the owner of the project want him self as creator this is the reason why i have add me as contributors

File: websites/N/Night.design/presence.ts (hunk 3 above)
Comment (Nightcorelena) [reply websites/N/Night.design/presence.ts]: I have remove it.

File: websites/N/Night.design/presence.ts (hunk 4 above)
Comment (Nightcorelena) [reply websites/N/Night.design/presence.ts]: Updated.

File: websites/N/Night.design/metadata.json (hunk 1 above)
Comment (EncryptedDev) [reply websites/N/Night.design/metadata.json:12]: That's not how it works

File: websites/N/Night.design/metadata.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+	"url": "night.design",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/4mDvKNa.png",
+	"thumbnail": "https://i.imgur.com/AbDSDug.png",
```
Comment (theusaf) [inline websites/N/Night.design/metadata.json:20]: The thumbnail image was removed or the link is invalid.

File: websites/N/Night.design/presence.ts [hunk 6]
```diff
@@ -0,0 +1,82 @@
+const presence = new Presence({
+	clientId: "1031730262092042321",
+});
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "logo",
```
Comment (theusaf) [inline websites/N/Night.design/presence.ts:7]: Use Imgur links over Discord asset names.
```suggestion
		largeImageKey: "https://i.imgur.com/4mDvKNa.png",
```

File: websites/N/Night.design/presence.ts [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+	presenceData.startTimestamp = Math.floor(Date.now() / 1000);
+	presenceData.buttons = [
+		{
+			label: "View Website",
+			url: "https://night.design/",
+		},
+	];
+
+	const domain = document.location.pathname.slice(3);
```
Comment (theusaf) [inline websites/N/Night.design/presence.ts:18]: Somewhat confusing variable name. `path` or `pathname` might be a better one.

File: websites/N/Night.design/presence.ts (hunk 3 above)
Comment (theusaf) [reply websites/N/Night.design/presence.ts]: > I have remove it.

In addition to the "actually page", [buttons to main pages are prohibited](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md#presencets).

File: websites/N/Night.design/presence.ts (hunk 7 above)
Comment (Nightcorelena) [reply websites/N/Night.design/presence.ts]: > Somewhat confusing variable name. `path` or `pathname` might be a better one.

I have change it to pathname

**Comment by theusaf** [comment]
* Resolve conversations you have completed
* Resolve lint errors

**Comment by Nightcorelena** [comment]
> * Resolve conversations you have completed
> * Resolve lint errors

I have try my best and fixed the most but i don't understand the last error message
"Error:   14:2  error    Use `switch` instead of multiple `else-if`  unicorn/prefer-switch"
I'm not sure home can I fix it

**Comment by RisingSunLight42** [comment]
> > * Resolve conversations you have completed
> > * Resolve lint errors
> 
> I have try my best and fixed the most but i don't understand the last error message
> "Error:   14:2  error    Use `switch` instead of multiple `else-if`  unicorn/prefer-switch"
> I'm not sure how can I fix it 
> 
> 

By replacing your if statements by a switch

File: websites/N/Night.design/presence.ts [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+				const pic = document.querySelector(".img-fluid") as HTMLImageElement;
+				presenceData.largeImageKey = pic.src;
+				presenceData.details = document.querySelector(".m-b-10").innerHTML;
+			}
+			break;
+		case `/u/${document.location.pathname.split("/")[3]}/activities`:
+			{
+				const image = document.querySelectorAll(
+					".rounded-circle"
+				)[0] as HTMLImageElement;
```
Comment (theusaf) [inline websites/N/Night.design/presence.ts:74]: These can be simplified to something like the following.
```suggestion
				const image = document.querySelector<HTMLImageElement>(
					".rounded-circle"
				);
```

File: websites/N/Night.design/presence.ts [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+				presenceData.largeImageKey = image.src;
+				presenceData.details = `View ${
+					document.location.pathname.split("/")[3]
+				}'s activitys`;
+			}
+			break;
+		case `/u/${document.location.pathname.split("/")[3]}/resources`:
+			{
+				const image1 = document.querySelectorAll(
```
Comment (theusaf) [inline websites/N/Night.design/presence.ts:83]: Ditto. (and for the others like this)

**Review by theusaf — CHANGES_REQUESTED** [review]

**Comment by theusaf** [comment]
Please mark the conversations/suggestions you have implemented/completed as resolved.

File: websites/N/Night.design/presence.ts (hunk 9 above)
Comment (Nightcorelena) [reply websites/N/Night.design/presence.ts]: > Ditto. (and for the others like this)

I have change it now

File: websites/N/Night.design/presence.ts [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+			presenceData.details = "Apply to become a Modder";
+			break;
+		case "/legal/terms-of-service":
+			presenceData.details = "Read the Terms of Service from Night.design";
+			break;
+		case "/legal/privacy-policy":
+			presenceData.details = "Read the Privacy Policy from Night.design";
+			break;
+		case "/legal/imprint":
+			presenceData.details = "Read the Imprint from Night.design";
```
Comment (theusaf) [inline websites/N/Night.design/presence.ts:61]: > if and else-ifs should be put in a Record if they're all static

These are static and should use the Record syntax:
https://github.com/PreMiD/Presences/blob/236523980322a4053a45c691a1739eb50e2aea48/websites/G/Geotastic/presence.ts#L12-L42

File: websites/N/Night.design/presence.ts [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+					document.querySelector<HTMLImageElement>(".rounded-circle");
+				presenceData.largeImageKey = image.src;
+				presenceData.details = `View ${
+					document.location.pathname.split("/")[3]
+				}'s activitys`;
+			}
+			break;
+		case `/u/${document.location.pathname.split("/")[3]}/resources`:
+			{
+				const image1 =
```
Comment (theusaf) [inline websites/N/Night.design/presence.ts:82]: You probably don't need to use a different variable name for each image in these as you've wrapped them in their own scope block.

File: websites/N/Night.design/presence.ts [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+				}'s designs`;
+			}
+			break;
+		case `/u/${document.location.pathname.split("/")[3]}/connected-accounts`:
+			{
+				const image2 =
+					document.querySelector<HTMLImageElement>(".rounded-circle");
+				presenceData.largeImageKey = image2.src;
+				presenceData.details = `View ${
+					document.location.pathname.split("/")[3]
```
Comment (theusaf) [inline websites/N/Night.design/presence.ts:96]: You use `document.location.pathname.split("/")[3]` many times in this code. Consider putting it in a variable for conciseness.

File: websites/N/Night.design/presence.ts [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+
+	switch (pathname) {
+		case "/home":
+			presenceData.details = "View the Homepage";
+			break;
+		case "/resources":
+			presenceData.details = "Search...";
+			break;
+		case "/settings/account":
+			presenceData.details = "Change my Account Settings";
```
Comment (theusaf) [inline websites/N/Night.design/presence.ts:22]: The wording of the details is a bit strange.
For example, I think a better way to word this one would be "Changing Account Settings" (basically, using '-ing' verbs, and removing 'my')

File: websites/N/Night.design/metadata.json [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+	"color": "#0303fc",
+	"category": "other",
+	"tags": [
+		"minecraft",
+		"design",
+		"capes",
+		"bandanas",
+		"shields"
+	]
+}
```
Comment (theusaf) [inline websites/N/Night.design/metadata.json:24]: This file is slightly formatted incorrectly. Use `yarn format` to format it.

**Review by theusaf — CHANGES_REQUESTED** [review]

**Review by theusaf — CHANGES_REQUESTED** [review]
PR title does not match commit conventions.
Reminder: 
**Mark conversations that you have completed as resolved by pressing the "Resolve Conversation" button.**

File: websites/N/Night.design/presence.ts [hunk 15]
```diff
@@ -0,0 +1,116 @@
+const presence = new Presence({
+	clientId: "1031730262092042321",
+});
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "https://i.imgur.com/4mDvKNa.png",
+	};
+
+	presenceData.startTimestamp = Math.floor(Date.now() / 1000);
```
Comment (EncryptedDev) [inline websites/N/Night.design/presence.ts:10]: Move this into the PresenceData object above

File: websites/N/Night.design/presence.ts [hunk 16]
```diff
@@ -0,0 +1,116 @@
...
+
+	const pathname = document.location.pathname.slice(3);
+
+	switch (pathname) {
```
Comment (EncryptedDev) [inline websites/N/Night.design/presence.ts:14]:
```suggestion
	switch (document.location.pathname.slice(3)) {
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

**Pull request closed without merging**